src/main/java/edu/ksu/canvas/attendance/model/AttendanceSummaryModel.java

```diff
@@ -18,47 +18,56 @@
         private final long courseId;
         private final long sectionId;
         private final long studentId;
+        private final String SisUserId;
         private final String studentName;
         private final int sumMinutesMadeup;
         private final int remainingMinutesMadeup;
         private final int sumMinutesMissed;
         private final double percentCourseMissed;
         private final boolean dropped;
         private final int totalClassesMissed;
+        private final int totalClassesExcused;
         private final int totalClassesTardy;
+        private final int totalClassesPresent;
 
 
         public Entry(long courseId, long sectionId, long studentId,
-                     String studentName, boolean dropped, int sumMinutesMadeup, int remainingMinutesMadeup,
+                     String sisUserId, String studentName, boolean dropped, int sumMinutesMadeup, int remainingMinutesMadeup,
                      int sumMinutesMissed, double percentCourseMissed) {
 
             this.courseId = courseId;
             this.sectionId = sectionId;
             this.studentId = studentId;
+            this.SisUserId = sisUserId;
             this.studentName = studentName;
             this.dropped = dropped;
             this.sumMinutesMadeup = sumMinutesMadeup;
             this.remainingMinutesMadeup = remainingMinutesMadeup;
             this.sumMinutesMissed = sumMinutesMissed;
             this.percentCourseMissed = percentCourseMissed;
+            this.totalClassesExcused = -1;
             this.totalClassesMissed = -1;
             this.totalClassesTardy = -1;
+            this.totalClassesPresent = -1;
         }
 
         public Entry(long courseId, long sectionId, long studentId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00107)